aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,764 @@
+        data: types.FileLike,
+        chunk_by_dims: List[str] = [Dimensions.SpatialZ, Dimensions.SpatialY, Dimensions.SpatialX],
+        S: int = 0,
+        **kwargs
+    ):
+        # Run super init to check filepath provided
+        super().__init__(data, **kwargs)
+
+        # Store parameters needed for _daread
+        self.chunk_by_dims = chunk_by_dims
+        self.specific_s_index = S
+        lif = LifFile(filename=self._file)
+        #  _chunk_offsets is a list of ndarrays (only way I could deal with inconsistent scene shape)
+        self._chunk_offsets, self._chunk_lengths = LifReader._compute_offsets(lif=lif)
+
+    @staticmethod
+    def _is_this_type(buffer: io.BytesIO) -> bool:
+        """
+        Test the file that was provided to check that the header is consistent with this reader.
+
+        Parameters
+        ----------
+        buffer: io.BytesIO
+            This is the contents of the file path the LifReader was initialized with.
+
+        Returns
+        -------
+        True / False
+            True if it has the right header byte structure False if it does not.
+
+        """
+        with BufferReader(buffer) as buffer_reader:
+            header = buffer_reader.read_bytes(n_bytes=8)
+
+            # if the buffer is to short return false
+            if len(buffer_reader.endianness) < 2 or len(header) < 8:
+                return False
+            # check for the magic byte
+            if buffer_reader.endianness[0] != LifReader.LIF_MAGIC_BYTE and header[1] != LifReader.LIF_MAGIC_BYTE:
+                return False
+            # check for the memory byte, if magic byte and memory byte are present return true
+            if header[6] == LifReader.LIF_MEMORY_BYTE:
+                return True
+        return False
+
+    @staticmethod
+    def _dims_shape(lif: LifFile):
+        """
+        Get the dimensions for the opened file from the binary data (not the metadata)
+
+        Parameters
+        ----------
+        lif: LifFile
+
+        Returns
+        -------
+        list[dict]
+            A list of dictionaries containing Dimension / depth. If the shape is consistent across Scenes then
+            the list will have only one Dictionary. If the shape is inconsistent the the list will have a dictionary
+             for each Scene. A consistently shaped file with 3 scenes, 7 time-points
+            and 4 Z slices containing images of (h,w) = (325, 475) would return
+            [
+             {'S': (0, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ].
+            The result for a similarly shaped file but with different number of time-points per scene would yield
+            [
+             {'S': (0, 1), 'T': (0,8), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (1, 2), 'T': (0,6), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (2, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ]
+
+        """
```

Comment:
Have you tried running `make serve-docs`? Do all docstrings (esp this one render correctly?)

Reply:
I just tried this and the and the output in the log was as follows:
```
# Sphinx version: 3.0.0b1
# Python version: 3.7.7 (CPython)
# Docutils version: 0.15.2 release
# Jinja2 version: 2.11.1
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/sphinx/cmd/build.py", line 279, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/sphinx/application.py", line 244, in __init__
    self.setup_extension(extension)
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/sphinx/application.py", line 398, in setup_extension
    self.registry.load_extension(self, extname)
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/sphinx/registry.py", line 414, in load_extension
    metadata = setup(app)
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/m2r.py", line 652, in setup
    app.add_source_parser('.md', M2RParser)
  File "/Users/jamies/anaconda3/envs/aics/lib/python3.7/site-packages/sphinx/application.py", line 1069, in add_source_parser
    self.registry.add_source_parser(*args, **kwargs)
TypeError: add_source_parser() takes 2 positional arguments but 3 were given
```
I'm fairly sure the comment above works as it's a modified form of a comment in the CZI reader that does work.

Reply:
You should merge master. `sphinx` released a new version and it broke our doc building, already fixed on master.